core/src/main/java/org/apache/carbondata/core/datamap/Segment.java

```diff
@@ -39,6 +41,11 @@
 
   private String segmentFileName;
 
+  /**
+   * List of tasks which are already got filtered through CG index operation.
+   */
+  private Set<String> filteredTaskNames = new HashSet<>();
```

Comment:
Instead of `taskName`, can we give a more formal one. I can suggest two: `indexShardName`, `segmentIndexName`

Reply:
ok